Components/Sensors/BME280.cs

```diff
@@ -197,20 +200,26 @@ private double ProcessTemperature(int TempInternal)
         /// <returns> Pressure reading in 1/256 increments, in Pa. </returns>
         private double ProcessPressure(int RawPress, int TempComp)
         {
-            long Var1, Var2, P;
-            Var1 = ((long)TempComp) - 128000;
-            Var2 = Var1 * Var1 * this.CompParams.P6;
-            Var2 = Var2 + ((Var1 * this.CompParams.P5) << 17);
-            Var2 = Var2 + (((long)this.CompParams.P4) << 35);
-            Var1 = ((Var1 * Var1 * this.CompParams.P3) >> 8) + ((Var1 * this.CompParams.P2) << 12);
-            Var1 = (((((long)1) << 47) + Var1)) * (this.CompParams.P1) >> 33;
-            if (Var1 == 0) { return 0; }
-            P = 1048576 - RawPress;
-            P = (((P << 31) - Var2) * 3125) / Var1;
-            Var1 = ((this.CompParams.P9) * (P >> 13) * (P >> 13)) >> 25;
-            Var2 = ((this.CompParams.P8) * P) >> 19;
-            P = ((P + Var1 + Var2) >> 8) + (((long)this.CompParams.P7) << 4);
-            return P / 256.000D;
+            double Var1, Var2, Var3, Press;
+            Var1 = (TempComp / 2.0) - 64000.0;
+            Var2 = Var1 * Var1 * this.CompParams.P6 / 32768.0;
+            Var2 = (Var2 + Var1) * this.CompParams.P5 * 2.0;
+            Var2 = (Var2 / 4.0) + (this.CompParams.P4 * 65536.0);
+            Var3 = this.CompParams.P3 * Var1 * Var1 / 524288.0;
+            Var1 = (Var3 + (this.CompParams.P2 * Var1)) / 524288.0;
+            Var1 = (1.0 + (Var1 / 32768.0)) * this.CompParams.P1;
+            if (Var1 > 0)
+            {
+                Press = 1048576.0 - RawPress;
+                Press = (Press - (Var2 / 4096.0)) * 6250.0 / Var1;
+                Var1 = this.CompParams.P9 * Press * Press / 2147483648.0;
+                Var2 = Press * this.CompParams.P8 / 32768.0;
+                Press = Press + ((Var1 + Var2 + this.CompParams.P7) / 16.0);
+                if (Press < 30000) { return 30000; }
+                if (Press > 110000) { return 110000; }
+            }
+            else { return 0; }
+            return Press;
```

Comment:
Have you actually tested this (aside from the UnitTests)?

Reply:
Yup, I'm using it in an indoor environmental monitor here at home. The readings seem pretty correct to me.